userland/shell/audit_log.c

```
@@ -0,0 +1,173 @@
+#include "shell_history_asm.h"
+#endif
+#include <errno.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static pthread_mutex_t g_audit_mutex = PTHREAD_MUTEX_INITIALIZER;
+static fl_log_sink_t *g_audit_sink;
+
+void fl_audit_set_sink(fl_log_sink_t *sink) { g_audit_sink = sink; }
```

Cppcheck (2.20.0)

[style] The function 'fl_audit_set_sink' is never used.

(unusedFunction)